### PR TITLE
MES-1120 Updated Test Slots

### PR DESCRIPTION
**Pull request by Cavalier07**
## Description and relevant Jira numbers

Implementation of the new slot designs for the following categories

A, A1, A2, AM, B, B1, B+E, C, C1, C1+E, C+E, D, D1, D1+E, D+E 

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [x] Tested by QA
- [x] PO's approval

unadjustednonraw_thumb_33

unadjustednonraw_thumb_34

### CODE REVIEW

File: src/pages/journal/components/test-outcome/test-outcome.scss [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+  }
+  .start-test-button {
+    background: #00823B;
+    box-shadow: inset 0 -2px 0 0 #006435;
+    border-radius: 10px 10px 10px 10px;
+  }
+  .submit-test-button {
+    background: #F5A623;
+    box-shadow: 0 2px 0 0 rgba(0, 0, 0, 0.5);
+    border-radius: 10px 10px 10px 10px;
```
Comment (szabolcs-keresztes) [inline src/pages/journal/components/test-outcome/test-outcome.scss]: `border-radius: 10px;`

File: src/pages/journal/components/test-slot/test-slot.ts [hunk 2]
```diff
@@ -10,9 +11,37 @@ export class TestSlotComponent implements SlotComponent {
   @Input()
   slot: any;
 
-  constructor() {}
+  vehicleDetails = {
```
Comment (szabolcs-keresztes) [inline src/pages/journal/components/test-slot/test-slot.ts]: Could go to a constants file

File: src/pages/journal/components/test-outcome/test-outcome.scss [hunk 3]
```diff
@@ -1,3 +1,15 @@
 test-outcome {
+  .outcome {
 
+  }
+  .start-test-button {
+    background: #00823B;
+    box-shadow: inset 0 -2px 0 0 #006435;
+    border-radius: 10px 10px 10px 10px;
```
Comment (szabolcs-keresztes) [inline src/pages/journal/components/test-outcome/test-outcome.scss]: Also here

File: src/pages/journal/components/test-slot/test-slot.ts (hunk 2 above)
Comment (Cavalier07) [reply src/pages/journal/components/test-slot/test-slot.ts]: Fixed

File: src/pages/journal/components/test-outcome/test-outcome.scss (hunk 3 above)
Comment (Cavalier07) [reply src/pages/journal/components/test-outcome/test-outcome.scss]: Fixed

File: src/pages/journal/components/test-outcome/test-outcome.scss [hunk 4]
```diff
@@ -1,3 +1,15 @@
 test-outcome {
+  .outcome {
 
+  }
+  .start-test-button {
+    background: #00823B;
```
Comment (johnmorleybjss) [inline src/pages/journal/components/test-outcome/test-outcome.scss]: Perhaps put these colours in to the colors map?

File: src/pages/journal/components/test-slot/test-slot.html [hunk 5]
```diff
@@ -1,21 +1,54 @@
...
     <ion-col col-2>
-      <time [time]="slot.slotDetail.start" [testComplete]="testComplete"></time>
+      <time
+        [time]="slot.slotDetail.start"
+        [testComplete]="testComplete"
+      >
+    </time>
     </ion-col>
-    <ion-col col-5>
+    <ion-col col-4 style="margin-left: -30px;">
```
Comment (johnmorleybjss) [inline src/pages/journal/components/test-slot/test-slot.html]: Would be nice to get this style out of the template.

File: src/pages/journal/components/test-slot/test-slot.html [hunk 6]
```diff
@@ -1,21 +1,54 @@
...
+            <test-category-icon
+              [category]="slot.booking.application.testCategory"
+            >
+            </test-category-icon>
+          </ion-col>
+        </ion-row>
+        <ion-row>
+          <ion-col no-padding>
+            <vehicle-details
+              [height]="slot.booking.application.vehicleHeight"
```
Comment (johnmorleybjss) [inline src/pages/journal/components/test-slot/test-slot.html]: Might be worth extracting a domain object to encapsulate these vehicle details?

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  }));
+
+  describe('Class', () => {
+    it('should create', () => {
+      expect(component).toBeDefined();
+    });
+  });
+
+  describe('DOM', () => {
+    it('should show vehicle details when showVehicleDetails is true', () => {
```
Comment (johnmorleybjss) [inline src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: This test name doesn't look quite right. You're setting `showDimensions`.

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      component.showDimensions = false;
+      component.showNumberOfSeats = true;
+      fixture.detectChanges();
+      const ionCols = fixture.debugElement.queryAll(By.css('ion-col'));
+      expect(ionCols).toHaveLength(2)
+    });
+
+    it('should not show number of seats when showNumberOfSeats is false', () => {
+      component.showDimensions = false;
+      component.showNumberOfSeats = true;
```
Comment (johnmorleybjss) [inline src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: Not doing what the test name says here - seems identical to the test above.

File: src/pages/journal/components/test-slot/test-slot.scss [hunk 9]
```diff
@@ -1,2 +1,10 @@
 test-slot {
+  ion-card {
+    &.test-slot-portrait-mode {
```
Comment (johnmorleybjss) [inline src/pages/journal/components/test-slot/test-slot.scss]: Is this to get 7 slots to fit without scrolling? I was under the impression we were going to allow scrolling in portrait?

Comment (Cavalier07) [reply src/pages/journal/components/test-slot/test-slot.scss]: Partly, it is also to try and keep a consistent height for all portrait slots as the data varies from 1 - 3 lines depending on category. On landscape we have less room and only use 2 lines so we keep the slots smaller.

File: src/pages/journal/components/test-outcome/test-outcome.scss (hunk 4 above)
Comment (Cavalier07) [reply src/pages/journal/components/test-outcome/test-outcome.scss]: done

File: src/pages/journal/components/test-slot/test-slot.html (hunk 5 above)
Comment (Cavalier07) [reply src/pages/journal/components/test-slot/test-slot.html]: done

File: src/pages/journal/components/test-slot/test-slot.html (hunk 6 above)
Comment (Cavalier07) [reply src/pages/journal/components/test-slot/test-slot.html]: I'm going to leave this one for now - there is the potential we are pulling this data out next sprint as part of the Journal-Slot-Rework ticket, I'll add it to the ticket to update it depending on which way we swing.

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts (hunk 7 above)
Comment (Cavalier07) [reply src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: fixed :)

File: src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts (hunk 8 above)
Comment (Cavalier07) [reply src/pages/journal/components/vehicle-details/__tests__/vehicle-details.spec.ts]: fixed - that's poor copy and paste work

File: src/components/test-category-icon/test-category-icon.ts [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+@Component({
+  selector: 'test-category-icon',
+  templateUrl: 'test-category-icon.html'
+})
+export class TestCategoryIconComponent {
+  @Input()
+  category: string;
+
+  categoryIcons = {
```
Comment (ammarhaiderbjss) [inline src/components/test-category-icon/test-category-icon.ts]: Could we use a string [Enum](https://www.typescriptlang.org/docs/handbook/enums.html) for the categories?

File: src/components/test-category-icon/test-category-icon.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    'A': 'assets/icon/test-categories/A.png',
```
Comment (ammarhaiderbjss) [inline src/components/test-category-icon/test-category-icon.ts]: Would it be better to have something like `iconPath` coming from config and then a function here to resolve the filename, e.g.
`(category) => \`${iconPath}/${category}.png\``

Comment (Cavalier07) [reply src/components/test-category-icon/test-category-icon.ts]: We aren't going to have 1 icon per category, once we get the full list each icon will be used multiple times, so unless we duplicate the images this wouldn't work

File: src/components/test-category-icon/test-category-icon.ts (hunk 10 above)
Comment (Cavalier07) [reply src/components/test-category-icon/test-category-icon.ts]: Done

**Pull request merged**